### PR TITLE
Add support for LLVM-14 and LLVM-15

### PR DESCRIPTION
**Pull request by kagayaku-hou**

### CODE REVIEW

File: include/SVF-FE/LLVMUtil.h [hunk 1]
```diff
@@ -104,6 +104,15 @@ inline bool isHeapAllocExtFunViaArg(const SVFFunction* fun)
     return fun && ExtAPI::getExtAPI()->is_arg_alloc(fun);
 }
 
+static inline Type *getPtrElementType(const PointerType* pty) {
+#if (LLVM_VERSION_MAJOR < 14)
+    return getPtrElementType(pty);
```
Comment (yuleisui) [inline include/SVF-FE/LLVMUtil.h:109]: This is a recursive call. Do you want `pty>getElementType()'

File: include/SVF-FE/LLVMUtil.h [hunk 2]
```diff
@@ -104,6 +104,15 @@ inline bool isHeapAllocExtFunViaArg(const SVFFunction* fun)
...
+#else
+    assert(!pty->isOpaque() && "Opaque Pointer is used, please recompile the source adding '-Xclang -no-opaque-pointer'");
+    return pty->getNonOpaquePointerElementType();
```
Comment (yuleisui) [inline include/SVF-FE/LLVMUtil.h:112]: Do we have any other way to return a correct type rather than an assertion here?

**Comment by yuleisui** [comment]
Thanks a lot for the contribution. Please find my comments.

File: include/SVF-FE/LLVMUtil.h (hunk 2 above)
Comment (kagayaku-hou) [reply include/SVF-FE/LLVMUtil.h:112]: With opaque pointers introduced in llvm-14, pointer type does not contain  element type any more. If you need the element type info, you need to compile the source and disable opaque pointer feature. Please read the following link:

https://llvm.org/docs/OpaquePointers.html

Comment (yuleisui) [reply include/SVF-FE/LLVMUtil.h:112]: I see. Thanks for the explanation.

**Comment by kagayaku-hou** [comment]
What is the status for this commit?

**Comment by yuleisui** [comment]
> What is the status for this commit?

This will be a big change because SVF-example, SVF-Teaching, and Test-suites are using LLVM-13.0. We will need to refresh all these to make sure the patch works for all.

**Pull request merged**